mlir/lib/Dialect/Rock/utility/transformMapUtils.cpp

```diff
@@ -699,6 +685,63 @@ int64_t mlir::rock::getMaxVectorization(ArrayAttr transforms, uint32_t dim,
   return result;
 }
 
+ArrayAttr mlir::rock::collapseContiguousMerges(ArrayAttr transforms,
+                                               ArrayRef<int64_t> outputShape) {
+  auto contigousMerges = findContiguousGroups(transforms, outputShape);
+  SmallVector<Attribute> newTransformMaps;
+  for (auto map : transforms.getAsRange<TransformMapAttr>()) {
+    bool changed = false;
+    SmallVector<TransformAttr> ops;
+    ops.reserve(map.getOps().size());
+    for (TransformAttr op : map.getOps()) {
+      if (op.getType() != TransformType::Merge) {
+        ops.push_back(op);
+        continue;
+      }
+      auto mergeData = contigousMerges.find({map, op});
```

Comment:
here too.

Reply:
This one's an iterator type, I think it's justified. The previous auto isn't have fixed it.